pkgs/http_profile/lib/http_profile.dart

```diff
@@ -82,105 +137,154 @@ final class HttpProfileRequestData {
   ///
-  /// Usage example:
+  /// For example:
   ///
   /// ```dart
-  /// profile.requestData.cookies = [
-  ///   'sessionId=abc123',
-  ///   'csrftoken=def456',
-  /// ];
+  /// // Foo: Bar
+  /// // Foo: Baz
+  ///
+  /// profile?.requestData.headersListValues({'Foo', ['Bar', 'Baz']});
   /// ```
-  set cookies(List<String> value) {
-    _data['cookies'] = [...value];
-    _updated();
-  }
-
-  /// The error associated with a failed request.
-  set error(String value) {
-    _data['error'] = value;
-    _updated();
-  }
-
-  /// Whether automatic redirect following was enabled for the request.
-  set followRedirects(bool value) {
-    _data['followRedirects'] = value;
-    _updated();
+  set headersListValues(Map<String, List<String>>? value) {
+    _checkAndUpdate();
+    if (value == null) {
+      _requestData.remove('headers');
+      return;
+    }
+    _requestData['headers'] = {...value};
   }
 
-  set headers(Map<String, List<String>> value) {
-    _data['headers'] = {...value};
-    _updated();
+  /// The request headers where duplicate headers are represented using
+  /// comma-seperated list of values.
+  ///
+  /// For example:
+  ///
+  /// ```dart
+  /// // Foo: Bar
+  /// // Foo: Baz
+  ///
+  /// profile?.requestData.headersCommaValues({'Foo', 'Bar, Baz']});
+  /// ```
+  set headersCommaValues(Map<String, String>? value) {
+    _checkAndUpdate();
+    if (value == null) {
+      _requestData.remove('headers');
+      return;
+    }
+    _requestData['headers'] = _splitHeaderValues(value);
   }
 
   /// The maximum number of redirects allowed during the request.
   set maxRedirects(int value) {
-    _data['maxRedirects'] = value;
-    _updated();
+    _checkAndUpdate();
+    _requestData['maxRedirects'] = value;
   }
 
   /// The requested persistent connection state.
   set persistentConnection(bool value) {
-    _data['persistentConnection'] = value;
-    _updated();
+    _checkAndUpdate();
+    _requestData['persistentConnection'] = value;
   }
 
   /// Proxy authentication details for the request.
   set proxyDetails(HttpProfileProxyData value) {
-    _data['proxyDetails'] = value._toJson();
-    _updated();
+    _checkAndUpdate();
+    _requestData['proxyDetails'] = value._toJson();
   }
 
-  const HttpProfileRequestData._(
+  HttpProfileRequestData._(
     this._data,
     this._updated,
   );
+
+  void _checkAndUpdate() {
+    if (_isClosed) {
+      throw StateError('HttpProfileResponseData has been closed, no further '
+          'updates are allowed');
+    }
+    _updated();
+  }
+
+  /// Signal that the request, including the entire request body, has been
+  /// sent.
+  ///
+  /// [bodySink] will be closed and the fields of [HttpProfileRequestData] will
+  /// no longer be changeable.
+  ///
+  /// [endTime] is the time when the request was fully sent. It defaults to the
+  /// current time.
+  void close([DateTime? endTime]) {
+    _checkAndUpdate();
+    _isClosed = true;
+    bodySink.close();
```

Comment:
Consider changing this and the other `bodySink.close()` lines to `unawaited(bodySink.close());`

Reply:
Done.